metric-collector/service-readiness/kubeflow-readiness.py

```diff
@@ -0,0 +1,140 @@
+import argparse
+from time import sleep, time
+
+import google.auth
+import google.auth.app_engine
+import google.auth.compute_engine.credentials
+import google.auth.iam
+from google.auth.transport.requests import Request
+import google.oauth2.credentials
+import google.oauth2.service_account
+from prometheus_client import start_http_server, Gauge
+import requests
+
+IAM_SCOPE = 'https://www.googleapis.com/auth/iam'
+OAUTH_TOKEN_URI = 'https://www.googleapis.com/oauth2/v4/token'
+METHOD = 'GET'
+KUBEFLOW_AVAILABILITY = Gauge('kubeflow_availability', 'Signal of whether IAP protected kubeflow is available')
+
+def metric_update(args, google_open_id_connect_token):
+  resp = requests.request(
+    METHOD, args.url,
+    headers={'Authorization': 'Bearer {}'.format(
+      google_open_id_connect_token)})
+  if resp.status_code == 200:
+    KUBEFLOW_AVAILABILITY.set(1)
+  else:
+    KUBEFLOW_AVAILABILITY.set(0)
+
+def main(unparsed_args=None):
+  parser = argparse.ArgumentParser(
+    description="Output signal of kubeflow service readiness.")
+
+  parser.add_argument(
+    "--url",
+    default="",
+    type=str,
+    help="kubeflow IAP-protected url")
+  parser.add_argument(
+    "--client_id",
+    default="",
+    type=str,
+    help="Service account json credential file")
+
+  args = parser.parse_args(args=unparsed_args)
+
+  if args.url == "":
+    sleep(2000)
+    return
+
+  # Figure out what environment we're running in and get some preliminary
+  # information about the service account.
+  bootstrap_credentials, _ = google.auth.default(
```

Comment:
nit: Why did you call it bootstrap_credentials and not just credentials? What is bootstrapping referring to?

Reply:
This name was used in IAP code examples, I think they were referring to fetch credential aggressively from environment and use whatever available (gcloud / service account key / ...).